Wikipedia/Code/UINavigationBar+WMFImageGalleryGradient.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import QuartzCore
+
+extension UINavigationBar {
+    
+    func wmf_applyGalleryTopGradientBackground() {
+        if (!CGRectIsNull(bounds)){
+            if (CGRectGetWidth(bounds) > 0 && CGRectGetHeight(bounds) > 0){
+                setBackgroundImage(wmf_galleryTopGradientBackgroundImage(), forBarMetrics: .Default)
+            }
+        }
+    }
+    
+    func wmf_galleryTopGradientBackgroundImage() -> UIImage {
+        UIGraphicsBeginImageContextWithOptions(bounds.size, false, UIScreen.mainScreen().scale)
```

Comment:
Nav bar tiles the background you provide. With a 1 pixel wide image wouldn't its pixels perforce be copied into the resulting pixel buffer *width* times?